app/javascript/elm/src/MobileSessionsFilters.elm

```diff
@@ -142,6 +195,68 @@ view model =
         ]
 
 
+viewParameterFilter : String -> Html Msg
+viewParameterFilter selectedParameter =
+    div []
+        [ h4 [] [ text "parameter" ]
+        , input
+            [ Attr.id "parameter-filter"
+            , clickWithoutDefault ShowSelectFormItemsPopup
+            , Attr.value selectedParameter
+            ]
+            []
+        ]
+
+
+viewPopup : (String -> Msg) -> Bool -> List String -> Html Msg
+viewPopup onSelect isPopupExtended items =
+    let
+        numberOfMainItems =
+            4
+
+        mainItems =
+            List.take numberOfMainItems items
+
+        moreItems =
+            List.drop numberOfMainItems items
+    in
+    div [ Attr.id "popup" ]
+        [ selectableItems mainItems onSelect
+        , if isPopupExtended then
+            div []
+                [ selectableItems moreItems onSelect
+                , button
+                    [ clickWithoutDefault TogglePopupState ]
+                    [ text "less parameters" ]
+                ]
+
+          else
+            button
+                [ clickWithoutDefault TogglePopupState ]
+                [ text "more parameters" ]
+        ]
+
+
+selectableItems : List String -> (String -> Msg) -> Html Msg
+selectableItems items onSelect =
+    items
+        |> List.map (\item -> li [] [ button [ Events.onClick (onSelect item) ] [ text item ] ])
+        |> ul []
+
+
+clickWithoutDefault : Msg -> Html.Attribute Msg
+clickWithoutDefault msg =
+    Events.custom "click" (Decode.map preventDefault (Decode.succeed msg))
```

Comment:
This overrides the default onClick which is defined in subscriptions ` Browser.Events.onClick (Decode.succeed ClosePopup)` - this way clicking anywhere on the page closes the popup, except for places where we want to override that